pyodide_build/buildpkg.py

```diff
@@ -40,15 +40,30 @@ def check_checksum(path, pkg):
 
 
 def download_and_extract(buildpath, packagedir, pkg, args):
-    tarballpath = buildpath / Path(pkg['source']['url']).name
-    if not tarballpath.is_file():
-        subprocess.run([
-            'wget', '-q', '-O', str(tarballpath), pkg['source']['url']
-        ], check=True)
-        check_checksum(tarballpath, pkg)
     srcpath = buildpath / packagedir
-    if not srcpath.is_dir():
-        shutil.unpack_archive(str(tarballpath), str(buildpath))
+
+    if 'url' in pkg['source']:
+        tarballpath = buildpath / Path(pkg['source']['url']).name
+        if not tarballpath.is_file():
+             subprocess.run([
+                  'wget', '-q', '-O', str(tarballpath), pkg['source']['url']
+             ], check=True)
+             check_checksum(tarballpath, pkg)
+
+        if not srcpath.is_dir():
+            shutil.unpack_archive(str(tarballpath), str(buildpath))
+
+    elif 'path' in pkg['source']:
+        srcdir = Path(pkg['source']['path'])
```

Comment:
Is the relative path relative to the file or to the home directory?

Reply:
Hi, the path currently is relative to the package directory where the meta.yaml is located.